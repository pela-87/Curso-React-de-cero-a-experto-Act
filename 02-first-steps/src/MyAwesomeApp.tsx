/* 
Tarea

1- Crear un componente de React dentro del src lladamo MyAwesomeApp
2- El componente debe de retornar (cambiar el nombre)
<h1>Fernando</h1>
<h3>Herrera</h3>

3- Hacer las configuraciones respectivas para poder ver el componente en pantalla.
*/

import type { CSSProperties } from "react";

const firstName = ' Jonatan';
const lastName = 'Villarruel';

const favoriteGames = ['Mortal Kombat', 'Bomber-man','Sonic'];
const isActive = true;

const address = {
    zipCode: 'ABC-123',
    country: 'Argentina'
}

/*
Se define como : CSSProperties para usar la interfase de react y
nos ofresca ayuda con las propiedades. Para eso se importa la interfase.
*/
const myStyles: CSSProperties = {
    backgroundColor: isActive ? 'blue' : 'red',
    borderRadius: 20,
    padding: 10,
    margin: 30,
}

export function MyAwesomeApp() {

    return (
        <div data-testid="div-app">
            <h1 data-testid="first-name-title">{ firstName }</h1>
            <h3>{ lastName }</h3>

            <p className="mi-class-favorita">{favoriteGames.join(', ')}</p>

            <p>{2 + 2}</p>
            <h1>{isActive ? 'Activo' : 'No activo'}</h1>

            {/* Esta es una forma de darle estilo */}
            {/* <p style={{
                backgroundColor: isActive ? 'blue' : 'red',
                borderRadius: 10,
                padding: 10,
            }}
            >{JSON.stringify(address)}</p> */}

            {/* esta es otra forma, mejor vista */}
            <p style={myStyles}>{JSON.stringify(address)}</p>
        </div>
    )
}