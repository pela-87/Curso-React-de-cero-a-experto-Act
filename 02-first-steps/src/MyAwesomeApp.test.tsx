import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';

import { MyAwesomeApp } from './MyAwesomeApp';


describe('MyAwesomeApp',() => {

    test('Should render firstName and lastName', () => {
        const { container } = render(<MyAwesomeApp />);
        //screen.debug();

        const h1 = container.querySelector('h1');
        const h3 = container.querySelector('h3');
        // console.log(h1.innerHTML);
        //expect(h1?.innerHTML).toBe('Jonatan');
        expect(h1?.innerHTML).toContain('Jonatan');
        expect(h3?.innerHTML).toContain('Villarruel');
        //Prueba para ver que falla y si se encuentra arriba, las otras pruebas no se ejecutan.
        //expect(1).toBe(2);
    });
    
    test('Should render firstName and lastName - screen', () => {
        render(<MyAwesomeApp />);
        screen.debug();

        // const h1 = screen.getByRole('heading', {
        //     level: 1,
        // });

        const h1 = screen.getByTestId('first-name-title');
        //console.log(h1.innerHTML);
        expect(h1?.innerHTML).toContain('Jonatan');        
    });

    test('should match snapshot', () => {
        const { container } = render(<MyAwesomeApp />);
        expect(container).toMatchSnapshot();
    });

    test('should match snapshot con screen', () => {
        render(<MyAwesomeApp />);
        expect(screen.getByTestId('div-app')).toMatchSnapshot();
    })
})