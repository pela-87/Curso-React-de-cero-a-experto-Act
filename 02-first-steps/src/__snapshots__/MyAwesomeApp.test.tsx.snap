// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
       Jonatan
    </h1>
    <h3>
      Villarruel
    </h3>
    <p
      class="mi-class-favorita"
    >
      Mortal Kombat, Bomber-man, Sonic
    </p>
    <p>
      4
    </p>
    <h1>
      Activo
    </h1>
    <p
      style="background-color: blue; border-radius: 20px; padding: 10px; margin: 30px;"
    >
      {"zipCode":"ABC-123","country":"Argentina"}
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snapshot con screen 1`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
     Jonatan
  </h1>
  <h3>
    Villarruel
  </h3>
  <p
    class="mi-class-favorita"
  >
    Mortal Kombat, Bomber-man, Sonic
  </p>
  <p>
    4
  </p>
  <h1>
    Activo
  </h1>
  <p
    style="background-color: blue; border-radius: 20px; padding: 10px; margin: 30px;"
  >
    {"zipCode":"ABC-123","country":"Argentina"}
  </p>
</div>
`;
